state/state.go

```diff
@@ -306,6 +308,32 @@ func (s *State) CacheAndUnpack(b *ui.Task, p *manifest.Package) error {
 	return nil
 }
 
+// This is needed as if you run CacheAndUnpack on x86_64 mac for a manifest
+// with an entry for arm64 mac packages it does on extract that as
+// isExtracted returns true.
+// This method will only cache the values and try to get a Digest
+func (s *State) CacheAndDontUnpack(b *ui.Task, p *manifest.Package) (string, error) {
+	actualDigest := ""
+	if !s.isCached(p) {
+		mirrors := append(p.Mirrors, s.generateMirrors(p.Source)...)
+		_, _, ch, err := s.cache.Download(b, p.SHA256, p.Source, mirrors...)
+		if err != nil {
+			return "", errors.WithStack(err)
+		}
+		actualDigest = ch
+	} else {
+		path := s.cache.Path(p.SHA256, p.Source)
```

Comment:
Is `p.SHA256` already the sha256 value? Can we use that instead of re-calculating?

Reply:
> > AFAIK, At the moment manifest files have sha256 and now this command is called locked-digests so these are the only two.
> 
> Then would it be more consistent to use "sha256" everywhere, not introducing the concepts of "checksum" or "digest"?

I have removed checksum. sha256 in the manifest file tells the user about what algorithm is used and that is essential information. lock-digests is a generic enough name that if we later want to use sha512 then it would still be applicable. wdyt ?

Reply:
I'm still not clear why we can't use `p.SHA256`?

Reply:
I think what @quad is suggesting is that if `p.SHA256 != ""`, just return that immediately.

Reply:
Those are all good points. AFAIK the p.SHA256 value is coming from a user and it was manually calculated using maybe something like:
```
sha256sum <file>
```
I am thinking if we should check the value here at least ?
The reason I am with that is we plan to use this tooling to automatically update digests in our sources so we should be trusting this more than user inputs ?

Reply:
made the change and skipping calculation if already present.